src/System.Security.Cryptography.X509Certificates/tests/CollectionTests.cs

```diff
@@ -228,7 +228,16 @@ public static void X509CertificateCollectionThrowsArgumentNullException()
                 Assert.Throws<ArgumentNullException>(() => ilist.Remove(null));
             }
 
-            Assert.Throws<ArgumentNullException>(() => new X509CertificateCollection.X509CertificateEnumerator(null));
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<NullReferenceException>(
+                    () => new X509CertificateCollection.X509CertificateEnumerator(null));
+            }
+            else
+            {
+                Assert.Throws<ArgumentNullException>(
+                    () => new X509CertificateCollection.X509CertificateEnumerator(null));
+            }
```

Comment:
There's an AssertExtensions.Throws overload that does this:
https://source.dot.net/#CoreFx.Private.TestUtilities/System/AssertExtensions.cs,109
```C#
AssertExtensions.Throws<ArgumentNullException, NullReferenceException>(() => => new X509CertificateCollection.X509CertificateEnumerator(null));
```